packages/debug/src/node/debug-adapter-factory.ts

```diff
@@ -66,6 +68,12 @@ export class LaunchBasedDebugAdapterFactory implements DebugAdapterFactory {
         const isForkOptions = (forkOptions: RawForkOptions | any): forkOptions is RawForkOptions =>
             !!forkOptions && !!forkOptions.modulePath;
 
+        if (!isForkOptions(executable)) {
+            if ((executable as DebugAdapterSpawnExecutable).command === 'node' && !environment.electron.is()) {
```

Comment:
From the [description](https://github.com/theia-ide/theia/pull/5492#issue-288783989):
> The node executable used to run Theia should also be used to run node-based debug adapters

This won't be the case if you run in electron.